models/remoteworkitem.go

```diff
@@ -0,0 +1,96 @@
+package models
+
+import (
+	"fmt"
+	"strconv"
+)
+
+const (
+	ProviderGithub string = "github"
+)
+
+// RemoteWorkItem represents a work item as it is stored in the database
+type RemoteWorkItem struct {
+	Fields map[string]interface{}
+}
+
+// WorkItemMap : holds the mapping between Remote key and local key in a WI
+type WorkItemMap map[string]interface{}
+
+// RemoteWorkItemSanitizer makes sure that the
+// Response from Provider API is sanitized into a
+// good looking flat 1-level dictionary
+type RemoteWorkItemMapper interface {
+	Flatten()
```

Comment:
No @aslakknutsen .
It's going to update the object member whose fields were an map[string]interface.

This is how it's being consumed:

```
githubRemoteWorkItem.Flatten()
```
Flatten() is actually optionally for XXXRemoteWorkItemType to implement. 
Example, in case of Github we already had everything **we need** as a flat 1-level dict